tests/sdk_tests/test_engine.py

```diff
@@ -54,10 +49,10 @@ def test_build_image(
     if generate_only:
         # Check stdout if it contains the correct docker build command
         assert got is None
-        command = f"docker buildx build --load --tag {image_name} --file {dockerfile_path} {tmpdir}"
+        command = f"docker buildx build --load --no-cache --tag {image_name} --file {dockerfile_path} {tmpdir}"
         assert command in caplog.text
     else:
-        assert got.attrs == mocked_docker.images.get(image_name).attrs
+        assert got.attrs == mocked_docker.get_image(image_name).attrs
```

Comment:
I think this PR would be most impactful (as in, increase re-usability of the CLI client) if we can minimize the diff in tests – if both objects quack ~ the same, we can switch back and forth between them, here and in other projects. Do you think it's feasible to adopt the `DockerClient` interface here, with `client.images.get` and `client.containers.get` instead of `client.get_image` / `client.get_container`?